### DISCUSSION TITLE
How to add cloudflare to existing project?

### DISCUSSION THREAD

**Discussion by putuoka** [remix-run/remix#5684] [Q&A]
i want to try host this example https://github.com/remix-run/examples/tree/main/_official-realtime-app into cloudflare pages.

i'm able to do that with fresh installation remix with "where do u want to deploy? cloudflare pages" option 

![image](https://user-images.githubusercontent.com/20880839/223315731-72153f17-349f-495c-9950-82c214e2d73c.png)

but how to do this with existing project for example above folder? i have tried but got an error on my cloudflare page dashboard

![image](https://user-images.githubusercontent.com/20880839/223316022-4a1fdfb8-e18d-4d5b-9a0c-9248761ea3f6.png)

**Comment by putuoka** [comment]
I have successfully make it working by replacing @remix-run/node to @remix-run/cloudflare
 as state on this issue
https://github.com/remix-run/remix/issues/3103

but the apps became not running properly

**Reply by kael** [reply]
Can you paste the errors logs ?

Reading the image of the logs above, it seems you're not running the required `NodeJS` version, you'd need to update to `NodeJS` 16.

Some resources for Remix Cloudflare integration:
- [Integrate Remix with Cloudflare Pages](https://www.jacobparis.com/guides/cloudflare-remix)
- [Serverless Side Rendering 2: Remix ](https://dev.to/kayis/serverless-side-rendering-2-remix-5e08)
- [Serverless Remix Sessions with Cloudflare Pages ](https://dev.to/fllstck/serverless-remix-sessions-with-cloudflare-pages-4lpo)
- [Deploy a Remix site](https://developers.cloudflare.com/pages/framework-guides/deploy-a-remix-site/)

**Reply by sergiodxa** [reply]
You can't use SSE in Cloudflare directly, you will have to integrate with Durable Objects and do more work, I remember the Cloudflare team has an example using Durable Objects for WebSockets, so it should be similar.
